java/ql/src/Security/CWE/CWE-023/PartialPathTraversal.qhelp

```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>A common way to check that a user-supplied path <code>SUBDIR</code> falls inside a directory <code>DIR</code>
+is to use <code>getCanonicalPath()</code> to remove any path-traversal elements and then check that <code>DIR</code>
+is a prefix. However, if <code>DIR</code> is not slash-terminated, this can unexpectedly allow accessing siblings of <code>DIR</code>.</p>
```

Comment:
Not sure about the English here, so perhaps suggesting:
```suggestion
is a prefix. However, if <code>DIR</code> is not slash-terminated, this can unexpectedly allow to access siblings of <code>DIR</code>.</p>
```